tsc/meetings/2019-08-29.md

```diff
@@ -0,0 +1,130 @@
+Minutes from 26th OpenVDB TSC meeting, July 25, 2019
+
+Attendees: *Nick* A., *Dan* B., *Jeff* L., *Ken* M., *Peter* C.
+
+Additional Attendees: Sean McDuffee
+
+Agenda:
+
+1) Quorum
+2) Secretary
+3) Round Robin
+4) Release Schedule for 6.2.0
+5) levelSetVolume
+6) VDB Smooth doesn't densify
+7) Bitscan/popcount
+8) Deprecation
+9) Combine SOP Performance
+10) Tree Refactoring
+11) Delayed Loading
+12) Next Meeting
+
+1) Quorum was confirmed.
+
+2) Secretary was Jeff Lait
+
+3) Round Robin
+
+A quick round robin to identify what is being worked on and what may
+make it for 6.2.
+
+Ken: Working on velocity extension.  Read-Only grid may end up being internal
+only.  Has found interesting results by making a write-only grid that allows
+better multithreaded scaling than usual multiple grids + merge.
+
+Nick: AX integration.  Feature branch is hoped to exist in a week or so.
+As well, CMake as required.  Should anistropic rasterization be higher
+priority?  In response to a later question, MPI will not be available with
+current priorities.
+
+Dan: Point partitioning improvements.  C++11 changes.  30-40% better performance
+on some datasets.  Merging of grids is open questions.  Likewise, refactoring
+tree hierarchy and improving combine SOP.
+
+Jeff: Hopes to get VDB Activate moved to OpenVDB.  6.1 integration into
+Houdini 18.0.
+
+Peter: Principle axis extraction from volumes, moment of inertia.  Sharpen
+filter.  Discussion on sharpen filter.  Does support non-level set, consensus
+was it should be kept regardless of differences with Ken's version.
+
+4) Release Schedule for 6.2.0
+
+No one had any critical items to be done for 6.2.0.
+
+5) levelSetVolume
+
+VDB Clip in 6.X leaves different values in inactive voxels, causing volume
+computation to change.  Consensus is that such half-open SDF don't have
+a proper volume so this is acceptable.
+
+Peter notes that there is an error computing volume of entirely empty
+leaf nodes as they have non-empty bounding boxes.
+
+6) VDB Smooth doesn't densify
+
+This is a known issue.  Jeff will submit a Jira ticket on it.
+
+Committee differs on whether the right course of action is to add a densify
+node to Houdini, or change all algorithms to support sparse tiles.
+
+However, algorithms that don't support sparse volumes should be documented as
+such, and the relevant help updated.
+
+7) Bitscan/popcount
+
+The TSC was unsure of the performance difference, so recommended asking
+the submitter to verify a measurable difference.  POPCOUNT is the more
+debatable as SSE 4.2 is more recent (Only with Houdini 17.0 did that become
+a requirement).
+
+We noted that this method does not belong in Tree, but belongs in Utils.
+No objections to moving to Utils for 7.
```

Comment:
I think this comment was about moving util/NodeMasks.h => tree/NodeMasks.h. Perhaps by retaining a dummy header for util/NodeMasks.h with a deprecated pragma.

Reply:
that's exactly right. Incidentally that's also the header file in which the intrinsic popcount was recently proposed (remember the response you recently wrote Jeff :)

Reply:
Okay, definitely my fault here.  My notes said that it was utils -> tree.  But I didn't believe that as it didn't make sense since it is a bunch of bit twiddling tools, so should be in utils.  So I assumed my notes were wrong!  Bad secretary!   I have no objections either way.